actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -759,6 +759,17 @@ def reject!(&block)
     end
     alias_method :delete_if, :reject!
 
+    # Returns a new instance of <tt>ActionController::Parameters</tt> with nil values removed.
+    def compact
+      new_instance_with_inherited_permitted_status(@parameters.compact)
+    end
+
+    # Removes all nil values in place and returns self.
+    def compact!
+      @parameters.compact!
+      self
```

Comment:
If we want to match `Hash#compact!`, I think this should be:

```suggestion
      self if @parameters.compact!
```

Reply:
Thanks, you're right!